src/breadcrumbs/breadcrumbs.scss

```diff
@@ -28,55 +28,74 @@
   // #region Component overrides
   > a {
     @include iui-anchor;
-    max-width: 24ch;
-    margin: 0 ($iui-s + 1);
+    padding: 0 $iui-s;
     overflow: hidden;
     white-space: nowrap;
     text-overflow: ellipsis;
+
+    &:focus {
+      outline-offset: -1px;
+    }
   }
 
-  > .iui-button {
+  .iui-button {
     @include iui-button-borderless;
-    --_iui-button-label-character-count: 24ch;
+    border: none;
+
+    &-label {
+      overflow: hidden;
+      white-space: nowrap;
+      text-overflow: ellipsis;
+    }
   }
 
-  &:not(.iui-current) > .iui-button {
-    --_iui-button-label-color: var(--iui-color-foreground-primary);
-    --_iui-button-label-color-hover: var(--_iui-button-label-color);
+  > * {
+    max-width: 26ch;
+  }
+
+  &:not(.iui-current) > * {
+    &,
+    &:hover,
+    &:active {
+      --_iui-button-foreground-color: var(--iui-color-foreground-primary);
+    }
   }
   // #endregion Component overrides
 }
 
 @mixin iui-breadcrumbs-text {
-  margin: 0 ($iui-s + 1);
-  max-width: 24ch;
+  padding: 0 $iui-s;
   overflow: hidden;
   white-space: nowrap;
   text-overflow: ellipsis;
 
   @at-root a#{&} {
```

Comment:
Something I didn't realize before: we are styling _all_ anchor elements that have this class because our expected usage is that the current item consists of a `<span>`. This might not be ideal because there are two cases where the current item can be an anchor instead of `<span>`:
- if the current item is an `<a>` without a `href`. See [html spec](https://html.spec.whatwg.org/multipage/text-level-semantics.html#:~:text=If%20the%20a%20element%20has%20no%20href%20attribute%2C%20then%20the%20element%20represents%20a%20placeholder).
- (less common) if clicking on the current item is meant to refresh the page. See [w3c example](https://www.w3.org/TR/wai-aria-practices/examples/breadcrumb/index.html).

In both of these cases, I think we might want to style it differently from the rest of the items.

Reply:
We could update the anchor to style `<a>` without a `href`, something like this:
![Anchor links](https://user-images.githubusercontent.com/849817/161341284-6428c1cc-9946-4a7b-9c6d-786900c5366e.png)

Reply:
Um, no this is not really about the anchor component. It's a specific use-case only for breadcrumbs

Reply:
I have this part ready to go:
```scss
> a {
  ...

  &:not([href]) {
    &,
    &:hover {
      text-decoration: none;
      cursor: default;
      @include themed {
        color: t(iui-text-color);
      }
    }
  }
}
```
But I'm struggling to remove the underline on the high contrast themes.
![Screen Shot 2022-04-15 at 8 28 37 AM](https://user-images.githubusercontent.com/849817/163570718-1a67b86c-3674-4064-b910-19b00b670f05.png)

Reply:
Hmm, we might need to target the high-contrast classes again if we want to remove the underline.

Also, maybe the [`:any-link`](https://developer.mozilla.org/en-US/docs/Web/CSS/:any-link) selector would help with the specificity here.

Reply:
So now it only works with `<a>`? Doesn't that break the existing approach of using `<span>`?

Also what about the non-override section? Will need to make it consistent there too.

Reply:
`<a>` with no `href` works.
`<span class="iui-breadcrumbs-text">` works.
`<span>` with no class does _not_ work currently.

Reply:
Do we want `<span>` with no class to work?

Reply:
It should work with anything passed under breadcrumb item, IMO..

Reply:
It now works with `<span>`.